runHarness/AppInstance/AppInstance.pm

```diff
@@ -751,7 +751,11 @@ sub waitForServicesRunning {
 				my $servicesRef = $self->getAllServicesByType($serviceType);
 				if ($#{$servicesRef} >= 0) {
 					foreach my $serviceRef (@$servicesRef) {
-						$allIsRunning &= $serviceRef->isRunning($logFile);
+						my ($serviceIsRunning, $errorStr) = $serviceRef->isRunning($logFile);
```

Comment:
Also, have you tested to be sure that $serviceIsRunning gets the correct value if kubernetesAreAllPodRunningWithNum returns just 0 or 1, rather than (0, "FailedScheduling")?  I assume this works, but I can't find any definitive statement of what happens in Perl when you assign a scalar to a list of scalars.

Reply:
Actually, with respect to my first comment, I guess that this will work because in the Docker case $errorStr will be UnDef and the method will work.  I am changing to approve.

Reply:
The cases that I have tested so far have all worked and I have been running with these changes for a number of days (mostly in cases where there are no FailedScheduling issues thus would return the simpler value).  I'll merge this so we can get more testing in on the 2.0-dev tree.